src/server/src/tokens.ts

```diff
@@ -1,58 +1,56 @@
+
+export {}
+
 const Nacl =       require('tweetnacl/nacl')
 const Nano =       require('nanocurrency')
 const Wallet =     require('nanocurrency-web')
 const Fs =         require('fs')
 const Tools =      require('./tools')
-const log_levels = {none:"none", warning:"warning", info:"info"}
+// const log_levels = {none:"none", warning:"warning", info:"info"}
 
 const API_TIMEOUT = 10000 // 10sec timeout for calling http APIs
 
-var work_server = "http://127.0.0.1:7000" // the work server for doing PoW (the node can be used as well, for example http://127.0.0.1:7076, but enable_control is needed in the node config)
-var token_price = 0.0001          // Nano per token
-var payment_timeout = 120         // timeout after 120sec
-var pending_interval = 2          // time to wait for each check for pending Nano
-var pending_threshold = "1"       // only allow pending tx above this raw value
-var pending_count = 10            // max number of pending to process per account for each order (normally only 1 should be needed)
-var difficulty_multiplier = "1.0" // Multipliers used when using the node for PoW
-var payment_receive_account = "nano_3jsonxwips1auuub94kd3osfg98s6f4x35ksshbotninrc1duswrcauidnue" // where to send the payment
-var min_token_amount = 1          // min allowed tokens to be purchased
-var max_token_amount = 10000000    // max allowed tokens to be purchased
-var log_level = log_levels.none     // the log level to use (startup info is always logged): none=zero active logging, warning=only errors/warnings, info=both errors/warnings and info
-
-// Read settings from file
+const loadSettings: () => TokenSettings = () => {
+  const defaultSettings: TokenSettings = {
+    work_server: "http://127.0.0.1:7000",
+    token_price: 0.0001,
+    payment_timeout: 120,
+    pending_interval: 2,
+    pending_threshold: "1",
+    pending_count: 10,
+    difficulty_multiplier: "1.0",
+    payment_receive_account: "nano_3jsonxwips1auuub94kd3osfg98s6f4x35ksshbotninrc1duswrcauidnue",
+    min_token_amount: 1,
+    max_token_amount: 10000000,
+    log_level: "none",
+  }
+  // Read settings from file
 // ---
-try {
-  const settings = JSON.parse(Fs.readFileSync('token_settings.json', 'UTF-8'))
-  work_server = settings.work_server
-  token_price = settings.token_price
-  payment_timeout = settings.payment_timeout
-  pending_interval = settings.pending_interval
-  pending_threshold = settings.pending_threshold
-  pending_count = settings.pending_count
-  difficulty_multiplier = settings.difficulty_multiplier
-  payment_receive_account = settings.payment_receive_account
-  min_token_amount = settings.min_token_amount
-  max_token_amount = settings.max_token_amount
-  log_level = settings.log_level
-}
-catch(e) {
-  console.log("Could not read token_settings.json", e)
+  try {
+    const readSettings: TokenSettings = JSON.parse(Fs.readFileSync('token_settings.json', 'UTF-8'))
+    return {...defaultSettings, ...readSettings}

```

Comment:
This merges defaultSettings with readSettings, overwriting with values from readSettings.